security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -682,7 +682,9 @@ func (sc *SecretCache) isTokenExpired() bool {
 // Prior to sending the request, it also sleep random millisecond to avoid thundering herd problem.
 func (sc *SecretCache) sendRetriableRequest(ctx context.Context, csrPEM []byte,
 	providedExchangedToken string, connKey ConnKey, isCSR bool) ([]string, error) {
-	backOffInMilliSec := rand.Int63n(sc.configOptions.InitialBackoff)
+	randSource := rand.NewSource(time.Now().UnixNano())
+	newRand := rand.New(randSource)
```

Comment:
this isn't that cheap, I was playing around with similar code that did this for every EDS response and it was using ~10% of pilot CPU.

Why do we need a new seed on every request? Is it so that each nodeagent instance/each connection has a different jitter, and if that case should we do this per nodeagent/per connection instead of per request?

Reply:
I agree, I think we can make randSource a singleton for the nodeagent?

Reply:
Nice catch, @howardjohn ! Thanks! I have moved this randSource into SecretCache, so that it is only created once.